astronomer/providers/amazon/aws/operators/emr.py

```diff
@@ -50,6 +52,21 @@ def execute(self, context: Context) -> None:
             # max_tries is deprecated so instead of max_tries using self.max_polling_attempts
             polling_attempts = self.max_polling_attempts
 
+        query_state = hook.check_query_status(job_id)
+        if query_state in hook.SUCCESS_STATES:
+            self.log.info(
+                f"Try : Query execution completed. Final state is {query_state}"
```

Comment:
Isn't it a job that is submitted to EMR? If yes, can we please replace `Query` with `job` in all the logs here? 

This I believe also calls for renaming Airflow OSS upstream hook methods to contain `job` instead of `query` in their name as we're essentially check job status and not query status. 
But, at least let us use `job` instead of `query` in our logs. WDYT?

Reply:
separate PR should be ok. @pankajkoti ??

Reply:
I think we can include this change as part of this PR. But I am okay if it comes as a separate PR but we should log the ticket.

Reply:
I am okay with a separate PR as long as we log a ticket for this cleanup. I would have preferred it to be in this PR  at least for the changes we're making in this PR as we're adding new logs with wrong information. But you feel this leads to inconsistencies in the logs between deferral and non-deferral, then yes, separate PR could make sense, but please create a ticket :) .

Reply:
I just created a separate issue and assigned it to myself. I'd like to close adding this check logic ASAP and work on the deferrable auto-switch earlier. But I think I will be able to work on this one during building the testing image for deferrable auto-swtich.

https://github.com/astronomer/astronomer-providers/issues/1105